CMakeLists.txt

```diff
@@ -481,18 +482,20 @@ endif(UNIX)
 set(SIMPLETESTS
    cargstest
    copyrecttest
+   wstest
 )
 
+set(wstest_add_src
+    ${TESTS_DIR}/wstestdata.inc
+   )
+
 if(CMAKE_USE_PTHREADS_INIT)
-  set(SIMPLETESTS
-      ${SIMPLETESTS}
-      encodingstest
-     )
+  list(APPEND SIMPLETESTS encodingstest)
 endif(CMAKE_USE_PTHREADS_INIT)
 
 file(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/test)
 foreach(t ${SIMPLETESTS})
-  add_executable(test_${t} ${TESTS_DIR}/${t}.c)
+  add_executable(test_${t} ${TESTS_DIR}/${t}.c ${${t}_add_src})
```

Comment:
SIMPLETESTS are called like that cause they have a single source, so better handler the ws test in  a separate case like tjunittest